### PR TITLE
Flesh out expectations, miscellaneous rewording

### PR DESCRIPTION
**Pull request by ml8**
# Goal

* I wanted to try to polish or make language more precise in some areas
* Build out the expectations and add examples where appropriate
* Clarify language where ambiguous (e.g., follow = adhere or follow = temporal relation)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,58 +1,91 @@
 # Mozilla Social Pull Request Review Guidelines
 
-PR reviews are we get to share and discuss code before we deliver features to our users. The review step should be meaningful, informative, interesting, and a key element of the continuous improvement of both our codebase and our relations with each other as a team.
+PR reviews are we get to share and discuss code before we deliver features to our users. The review step should be meaningful, informative, interesting, and a key element of the continuous improvement of both our codebase and our relationships with each other as a team.
```
Comment (efixler) [inline README.md:3]: 👍

File: README.md [hunk 2]
```diff
@@ -1,58 +1,91 @@
...
-- Collaboration and Feedback: Code reviews provide a platform for constructive feedback, discussions, and ideas that can lead to better solutions and continuous improvement.
-- Codebase Health: Code reviews can help us prevent technical debt and help identify both risky low-quality code, and areas for improvement in our developmebt process and tooling.
-- Onboarding and Knowledge Transfer: Reviews help us onboard new team members by exposing them to our codebases and practices.
+- __Maintain code quality and correctness:__ Code reviews help identify and address logic errors, bugs, code style departures, and security vulnerabilities. It ensures that code being merged meets our quality standards and helps us to evolve those standards.
+- __Knowledge sharing:__ Reviews are the main forum for discussions of real code and should facilitate elevated conversations across the team. PR reviews provide an opportunity for engineers to learn from one another, share best practices, and gain insights into the different parts of the codebase.
```
Comment (efixler) [inline README.md]: How about active form "Share Knowledge"?

File: README.md [hunk 3]
```diff
@@ -1,58 +1,91 @@
...
+- __Consistency:__ Reviews help ensure that the whole team follows the same conventions and coding standards, leading to a more readable and maintainable codebase.
+- __Collaboration and feedback:__ Code reviews provide a platform for constructive feedback, discussions, and ideas that can lead to better solutions and continuous improvement.
```
Comment (efixler) [inline README.md:17]: Should we bicker about title case?

File: README.md [hunk 4]
```diff
@@ -1,58 +1,91 @@
...
-- Spend the time to read, understand, and preferably test what's in the PR -- time spent here benefits all of us and our users.
-- Ask questions - don't assume that you know why the author made the choices they did.
-- Try to understand the big picture -- where this code is going, how it's meant to serve users and how it will fit into our codebase.
-- Use the PR forum for contructive discussion
-- Always consider perfomance, security, and adherence to existing mental models and patterns.
-- An approval means you're confident that this PR will both do what it's supposed to do, improve the Mozilla Social codebase, and not introduce technical debt.
-- Expect to complete your review within 1 business day of the PR going to review. If you can't do that, please tell the author early on.
-
-### Reviewer Expectations
+- __First and foremost: use the PR forum for _constructive_ discussion.__ Discuss
```
Comment (bassrock) [inline README.md:57]: Should we add something about following the [Mozilla Participation Guidelines ](https://www.mozilla.org/en-US/about/governance/policies/participation/) and to note that there are always multiple ways to code things.

File: README.md [hunk 5]
```diff
@@ -1,58 +1,91 @@
...
+  and critique __the code, not the author.__
+- Try to understand the big picture. Where this code is going? How is it meant to serve users? How will it fit into our codebase?
+- Spend the time to read, understand, and preferably test what's in the PR--time
+  spent here benefits all of us and our users. Don't rubber-stamp code, even if
+  it is a small change. Detailed and thorough code review improves our codebase
+  over time.
+- Consider performance, security, and adherence to existing mental models and patterns.
+- Ask questions. Don't assume that you know why the author made the choices they
```
Comment (bassrock) [inline README.md:67]: Should add something like `lean into the fact that the author had good intent` don't immediately assume bad.

File: README.md [hunk 6]
```diff
@@ -1,58 +1,91 @@
...
+  did. If you see an alternative approach, or might prefer code to be written
+  differently, __ask__ the author why they structured something the way that
+  they did--there might be a reason that you don't see.
+- If you require a change to be made or see an issue that should be addressed,
+  __explain__ the _whys_ and your suggested _hows_.
+
+  > __Unhelpful:__ "This is a bug; use `int` or at least make this `unsigned`."
```
Comment (bassrock) [inline README.md:76]: I 😍 these changes. These are great, be helpful and ask, not tell.

File: README.md [hunk 7]
```diff
@@ -1,58 +1,91 @@
...
-- PRs should be small both so that reviews are digestible and to reduce deployment risks
-- Code should conform to linting/style expectations
-- All functions/methods should have comment blocks in the standard documentation format for the language (generally docblock)
-- Unit test coverage is expected and reported
-- Commits follow semantic commits
+- PRs should be small. Meaningful reviews require digestible changes. Small changes also reduce deployment risks.
+- Code should conform to linting/style expectations.
+- All functions/methods should have comment blocks in the standard documentation format for the language (generally docblock).
+- Ideally, all code should be checked in alongside tests that validate its
+  correctness. These should include unit tests, at a minimum. Larger features
```
Comment (efixler) [inline README.md]: My observation across the industry in general is that integration tests tend to fall into unmaintenance because they are slow and/or the infra is brittle. My observation here is a bias towards integration tests over unit tests, which I'd like to nudge over towards unit tests.

I had kind of left the testing part intentionally vague for that reason. It's not a super-hard-formed stance but I wanted to pause on this for a second.

**Review by bassrock — COMMENTED** [review]
LGTM left a few random comments

File: README.md [hunk 8]
```diff
@@ -1,58 +1,91 @@
...
+  should include integration tests (possibly in parallel PRs, depending on
+  feature size).
+- Use semantic commit messages.
```
Comment (efixler) [inline README.md]: Ok, can we include the glossary of terms we use here if we're promoting this?

File: README.md [hunk 9]
```diff
@@ -1,58 +1,91 @@
...
-- Most reviews will require one 👍🏼
-- Initial reviewer is populated from an appropriate per-domain set of reviewers
-
-Some reviews will require 2 👍🏼, based on the characteristics of the review. Some of these include:
-
-- Does this PR introduce new infrastructure or significant new components?
-- Does this PR cross domains, teams or component areas?
-- Does the author want extra 👀 for any other reason, like complexity, questions, mentoring, etc.
+- An approval means you're confident that this PR will both do what it's supposed to do, improve the Mozilla Social codebase, and not introduce technical debt.
+- Expect to start your review within one business day of the PR going to review and complete it within another. If you can't do that, please tell the author early on.
```
Comment (efixler) [inline README.md:83]: I'm not sure I buy into 2-day turnarounds

Comment (ml8) [reply README.md:83]: wdym? The previous text was 1-day, do you mean you want shorter? I find 1 day total turnaround time a difficult SLO to meet if I'm going to give a meaningful review, especially if I am heads-down. My opinion is that encouraging fast reviews is a trade-off in review quality, so we should try to have some middle ground: ack and start the review within a day + finish it within a day.

File: README.md (hunk 8 above)
Comment (ml8) [reply README.md]: Yes -- do y'all have a recommended link? I'm not sure what your policies are? I'm just going to add my suggestions as a stab for commits, but we can/should link to standards if we're going to use them.

File: README.md (hunk 6 above)
Comment (ml8) [reply README.md:76]: Thanks!

File: README.md (hunk 5 above)
Comment (ml8) [reply README.md:67]: Great! Done!

File: README.md (hunk 4 above)
Comment (ml8) [reply README.md:57]: 100% yes, I added something at the top -- PTAL

File: README.md (hunk 7 above)
Comment (ml8) [reply README.md]: I agree. I think it depends on the scope of integration tests (e2e vs. component vs. ...). I'll back this out and add a note about forthcoming test requirements/philosophy.

File: README.md (hunk 3 above)
Comment (ml8) [reply README.md:17]: Haha up to you. I like the ACM style guide: all caps or initial caps for all words section titles, then only initial caps in subsection titles (incl. paragraphs). What do you prefer? 🤓

File: README.md (hunk 2 above)
Comment (ml8) [reply README.md]: SGTM!

File: README.md (hunk 3 above)
Comment (efixler) [reply README.md:17]: Let's do it your/ACM way :)

**Review by efixler — APPROVED** [review]
